server/webdriver/tests/vis_map_test.py

```diff
@@ -0,0 +1,259 @@
+from selenium.webdriver.support.ui import WebDriverWait
+
+from server.webdriver.base import WebdriverBaseTest
+import server.webdriver.shared as shared
+
+MAP_URL = '/tools/visualization#visType=map'
+URL_HASH_1 = '&place=geoId/06&placeType=County&sv=%7B"dcid"%3A"Count_Person_Female"%7D'
+PLACE_SEARCH_CA = 'California'
+
+
+# Class to test the map visualization tool.
+class TestVisMap(WebdriverBaseTest):
+
+  def test_server_and_page(self):
+    """Test the server can run successfully."""
+    self.driver.get(self.url_ + MAP_URL)
+
+    # Assert 200 HTTP code: successful page load.
+    req = urllib.request.Request(self.driver.current_url)
+    with urllib.request.urlopen(req) as response:
+      self.assertEqual(response.getcode(), 200)
+
+    # Assert 200 HTTP code: successful JS generation.
+    req = urllib.request.Request(self.url_ + '/visualization.js')
+    with urllib.request.urlopen(req) as response:
+      self.assertEqual(response.getcode(), 200)
+
+    # Assert page title is correct.
+    TITLE_TEXT = "Tools - Data Commons"
+    WebDriverWait(self.driver,
+                  self.TIMEOUT_SEC).until(EC.title_contains(TITLE_TEXT))
+    self.assertEqual(TITLE_TEXT, self.driver.title)
+
+    # Wait until the page has loaded.
+    element_present = EC.presence_of_element_located(
+        (By.CLASS_NAME, 'visualization-app'))
+    WebDriverWait(self.driver, self.TIMEOUT_SEC).until(element_present)
+
+    # Assert page heading and selected tab are correct.
+    page_header = self.driver.find_element(
+        By.XPATH, '//*[@id="main-pane"]/div/div[2]/div[1]/div[1]/h3')
+    self.assertEqual(page_header.text, 'Map Explorer')
+    selected_tab = self.driver.find_element(
+        By.CSS_SELECTOR, ".vis-type-selector .selected .label")
+    self.assertEqual(selected_tab.text, 'Map Explorer')
+
+  def test_charts_from_url(self):
+    """Given the url directly, test the page shows up correctly"""
+    self.driver.get(self.url_ + MAP_URL + URL_HASH_1)
+
+    # Wait until the chart has loaded.
+    WebDriverWait(self.driver, self.TIMEOUT_SEC).until(shared.charts_rendered)
+
+    # Assert place name is correct.
+    place_name_chip = self.driver.find_element(
+        By.CSS_SELECTOR, '.selected-option-chip.place .chip-content')
+    self.assertTrue('California' in place_name_chip.text)
+
+    # Assert place type is correct.
+    place_type_chip = self.driver.find_element(
+        By.CSS_SELECTOR, '.selected-option-chip.place-type .chip-content')
+    self.assertTrue('County' in place_type_chip.text)
+
+    # Assert stat var is correct.
+    stat_var_chip = self.driver.find_element(
+        By.CSS_SELECTOR, '.selected-option-chip.stat-var .chip-content')
+    self.assertTrue('Female Population' in stat_var_chip.text)
+
+    # Assert chart is correct.
+    chart_title = self.driver.find_element(
+        By.XPATH,
+        '//*[@id="main-pane"]/div/div[3]/div[2]/div[1]/div[1]/div/div[1]/h4')
```

Comment:
ditto here, and for a few more xpath blocks below

Reply:
done